src/main/java/vazkii/botania/api/recipe/IManaInfusionRecipe.java

```diff
@@ -25,15 +25,51 @@
 public interface IManaInfusionRecipe extends IRecipe<IInventory> {
 	ResourceLocation TYPE_ID = new ResourceLocation(BotaniaAPI.MODID, "mana_infusion");
 
+	/**
+	 * Attempts to match the recipe.
+	 *
+	 * @param stack The whole stack that is in the Mana Pool (when actually crafting)
+	 *              or in the player's hand (for the HUD).
+	 * @return Whether this recipe matches the given stack.
+	 */
 	boolean matches(ItemStack stack);
 
+	/**
+	 * Get the recipe output, used for display (in JEI or the HUD).
+	 * If {@link #getRecipeOutput(ItemStack)} isn't overridden, this is also the actual result of the crafting recipe.
+	 *
+	 * @return The output stack of the recipe.
+	 */
+	@Nonnull
+	@Override
+	ItemStack getRecipeOutput();
+
+	/**
+	 * Get the actual recipe output, not just for display. Defaults to a copy of {@link #getRecipeOutput()}.
+	 *
+	 * @param input The whole stack that is in the Mana Pool, not a copy.
+	 * @return The output stack of the recipe for the specific input.
+	 */
+	@Nonnull
+	default ItemStack getRecipeOutput(@Nonnull ItemStack input) {
+		return getRecipeOutput().copy();
+	}
+
+	/**
+	 * Get the block state that must be under the Mana Pool for this recipe, or {@code null} if it can be anything.
+	 *
+	 * @return The catalyst block state.
+	 */
```

Comment:
This javadoc needs an update as this is using state ingredients now

Reply:
wack angery